dev-tools/packaging/package_test.go

```diff
@@ -939,3 +946,69 @@ func readDockerManifest(r io.Reader) (*dockerManifest, error) {
 
 	return manifests[0], nil
 }
+
+func checkSha512PackageHash(t *testing.T, packageFile string) {
+	t.Run("check hash file", func(t *testing.T) {
+		expectedHashFile := packageFile + ".sha512"
+		require.FileExists(t, expectedHashFile, "hash file for package %q should exist with name %q", packageFile, expectedHashFile)
+
+		// calculate SHA512 hash for the file
+		hashFile, err := os.Open(expectedHashFile)
+		require.NoError(t, err, "hash file should be readable")
+
+		checksumsMap := readHashFile(t, hashFile)
+
+		packageBaseName := filepath.Base(packageFile)
+		require.Containsf(t, checksumsMap, packageBaseName, "checksum file should contain an entry for %q", packageBaseName)
+
+		// compare checksum entry with actual package hash
+		checksum := calculateChecksum(t, packageFile, sha512.New())
+
+		assert.Equalf(t, checksum, checksumsMap[packageBaseName], "checksum for file %q does not match", packageFile)
+	})
+
+}
+
+func calculateChecksum(t *testing.T, file string, hasher hash.Hash) string {
+
+	input, err := os.Open(file)
+	require.NoErrorf(t, err, "error opening input file %q", file)
+
+	defer func(input *os.File) {
+		errClose := input.Close()
+		assert.NoErrorf(t, errClose, "error closing input file %q", file)
+	}(input)
```

Comment:
Just out of curiosity, why don't you capture `input` directly in the closure?

Reply:
My IDE generates code this way when doing quick fixes for handling errors and it works for me... In general I prefer explicit arguments passed to a closure/anonymous function over "magic" variable capture from an outer scope